torchdata/dataloader2/reading_service.py

```diff
@@ -184,67 +181,57 @@ class PrototypeMultiProcessingReadingService(ReadingServiceInterface):
         multiprocessing_context (str, optional): Multiprocessing starting method.
             If method is None then the default context is returned.
             Otherwise, method should be 'fork', 'spawn'.
+        prefetch_worker: (int, 10 by default): Number of data will be prefetched at
+            the end of each worker process.
+        prefetch_mainloop: (int, 10 by default): Number of data will be prefetched
+            at the end of the whole pipeline in the main process.
+        worker_init_fn: (Callable, optional): Function to be called when each worker
+            process launches with ``DistInfo``, ``WorkerInfo`` and ``DataPipe``
+            as the expected arguments.
+        worker_reset_fn: (Callable, optional): Function to be called at the beginning
+            of each epoch in each worker process with ``DistInfo``, ``WorkerInfo``
+            and ``DataPipe`` as the expected arguments.
+
     """
     num_workers: int
+    multiprocessing_context: Optional[str]
+    prefetch_worker: int
+    prefetch_mainloop: int
+    worker_init_fn: Optional[Callable[[DataPipe, DistInfo, WorkerInfo], DataPipe]]
+    worker_reset_fn: Optional[Callable[[DataPipe, DistInfo, WorkerInfo], DataPipe]]
     processes: List
     datapipes: List
     end_datapipe: Optional[DataPipe]
     _mp: bool
     _pg: Optional[dist.ProcessGroup]
-    _world_size: int
-    _rank: int
+    _dist_info: DistInfo
 
     def __init__(
         self,
         num_workers: int = 0,
-        multiprocessing_context=None,
+        multiprocessing_context: Optional[str] = None,
```

Comment:
is this  `multiprocessing_context` expected to be one of the 'fork', 'spawn' and 'forkserver'  ?

Reply:
Not always. For windows, only `spawn` can be supported. That's the reason I checked the validity of `mp_context` using `mp.get_all_start_methods`.
See: https://docs.python.org/3/library/multiprocessing.html#multiprocessing.get_all_start_methods

Reply:
Right, i mean it cannot be out of these three options~ (although not all of them are supported~)

Reply:
You are right